cmd/devp2p/internal/ethtest/eth66_suiteHelpers.go

```diff
@@ -242,25 +242,37 @@ func sendSuccessfulTx66(t *utesting.T, s *Suite, tx *types.Transaction) {
 	time.Sleep(100 * time.Millisecond)
 	recvConn := s.setupConnection66(t)
 	// Wait for the transaction announcement
-	switch msg := recvConn.ReadAndServe(s.chain, timeout).(type) {
-	case *Transactions:
-		recTxs := *msg
-		if len(recTxs) < 1 {
-			t.Fatalf("received transactions do not match send: %v", recTxs)
-		}
-		if tx.Hash() != recTxs[len(recTxs)-1].Hash() {
-			t.Fatalf("received transactions do not match send: got %v want %v", recTxs, tx)
-		}
-	case *NewPooledTransactionHashes:
-		txHashes := *msg
-		if len(txHashes) < 1 {
-			t.Fatalf("received transactions do not match send: %v", txHashes)
-		}
-		if tx.Hash() != txHashes[len(txHashes)-1] {
-			t.Fatalf("wrong announcement received, wanted %v got %v", tx, txHashes)
+	for {
```

Comment:
How can this  possibly ever succeed? It will loop forever, or, in this case, eventually error out on when the remote peer don't send any more data:
```
unexpected message in sendSuccessfulTx: (*ethtest.Error)(could not read from connection: read tcp 127.0.0.1:45992->127.0.0.1:30303: i/o timeout)
```
You never exit the loop otherwise.

Reply:
So, what it's actually failing on is: 
```
wrong announcement received, wanted 0x7d58c33ec2867fd7f9631b31258468d8b1b9cf71f57672e12eacc4e5a63ed84a got [0x7d58c33ec2867fd7f9631b31258468d8b1b9cf71f57672e12eacc4e5a63ed84a 0x469d063bf70cc83df14ce191337642698da263d47d8003a7df581cdeadc2e601]

```
It does announce `0x7d58c33ec2867fd7f9631b31258468d8b1b9cf71f57672e12eacc4e5a63ed84a`, just not in the particular location where the test is looking :)